docs/src/ice_dynamics.md

```diff
@@ -152,12 +161,28 @@ constants_and_parameters = (
   stress_A = A)
 ```
 
-# Define functions
+## Define functions
 
 In order to solve our equations, we will need numerical linear operators that give meaning to our symbolic operators. In the DEC, there are a handful of operators that one uses to construct all the usual vector calculus operations, namely: ♯, ♭, ∧, d, ⋆. The CombinatorialSpaces.jl library specifies many of these for us.
 
-
 ``` @example DEC
+function create_average_matrix(sd)
+  I = Vector{Int64}()
+  J = Vector{Int64}()
+  V = Vector{Float64}()
+  for e in 1:ne(sd)
+      append!(J, [sd[e,:∂v0],sd[e,:∂v1]])
+      append!(I, [e,e])
+      append!(V, [0.5, 0.5])
+  end
+  avg_mat = sparse(I,J,V)
+end
+
+# TODO: Move this sharp matrix out of the generate
+function create_sharp_matrix(sd)
+
+end
```

Comment:
It's not done and I left the empty stub as a reminder. The idea is to refactor this code out into the function stub I created since this is all being run every sharp computation. https://github.com/AlgebraicJulia/Decapodes.jl/blob/85c70f1a3a837a68b9a5d2fd9b6f67640337cc07/docs/src/ice_dynamics.md?plain=1#L193-L204 If the doc building times are quick enough we could just come back to this when we have a real implementation of this 1D sharp rather than this bespoke one.

Reply:
If refactoring this will take longer than *today*, then I think we can merge this and you can open a new issue for this refactor to create a permanent 1D sharp. Just reading the loops, I don't like this implementation. I think there is a much more performant way to do build this matrix that is worth figuring out.

Reply:
Yeah that sounds good to me. If I remember correctly these docs didn't take too long to build so chances are I'll leave this alone to be refactored properly later.